framework/src/utils/FillBetweenPointVectorsTools.C

```diff
@@ -0,0 +1,655 @@
+  {
+    // This is invalid type #2
+    throw MooseException("This mesh generator does not work for the provided external boundary as "
+                         "it is not a closed loop.");
+    return false;
+  }
+  // It the boundary is a loop, check if azimuthal angles change monotonically
+  else
+  {
+    // Utilize cross product here.
+    // If azimuthal angles change monotonically,
+    // the z components of the cross products are always negative or positive.
+    std::vector<Real> ordered_node_azi_list;
+    for (unsigned int i = 0; i < boundary_ordered_node_list.size() - 1; i++)
+    {
+      ordered_node_azi_list.push_back(
+          (*mesh.node_ptr(boundary_ordered_node_list[i]) - origin_pt)
+              .cross(*mesh.node_ptr(boundary_ordered_node_list[i + 1]) - origin_pt)(2));
+      // Use this opportunity to calculate maximum radius
+      max_node_radius = std::max((*mesh.node_ptr(boundary_ordered_node_list[i]) - origin_pt).norm(),
+                                 max_node_radius);
+    }
+    std::sort(ordered_node_azi_list.begin(), ordered_node_azi_list.end());
+    if (ordered_node_azi_list.front() * ordered_node_azi_list.back() < 0.0)
+    {
+      // This is invalid type #3
+      throw MooseException("This mesh generator does not work for the provided external boundary "
+                           "as azimuthal angles of consecutive nodes do not change monotonically.");
+      return false;
+    }
+    else
+      return true;
+  }
+}
+
+bool
+isBoundarySimpleClosedLoop(ReplicatedMesh & mesh,
+                           Real & max_node_radius,
+                           const Point origin_pt,
+                           const boundary_id_type bid)
+{
+  std::vector<dof_id_type> dummy_boundary_ordered_node_list;
+  return isBoundarySimpleClosedLoop(
+      mesh, max_node_radius, dummy_boundary_ordered_node_list, origin_pt, bid);
+}
+
+bool
+isBoundarySimpleClosedLoop(ReplicatedMesh & mesh, const Point origin_pt, const boundary_id_type bid)
+{
+  Real dummy_max_node_radius;
+  return isBoundarySimpleClosedLoop(mesh, dummy_max_node_radius, origin_pt, bid);
+}
+
+bool
+isBoundaryOpenSingleSegment(ReplicatedMesh & mesh,
+                            Real & max_node_radius,
+                            std::vector<dof_id_type> & boundary_ordered_node_list,
+                            const Point origin_pt,
+                            const boundary_id_type bid)
+{
+  try
+  {
+    isBoundarySimpleClosedLoop(mesh, max_node_radius, boundary_ordered_node_list, origin_pt, bid);
+  }
+  catch (MooseException & e)
+  {
+    if (((std::string)e.what())
+            .compare("This mesh generator does not work for the provided external boundary as it "
+                     "is not a closed loop.") != 0)
+    {
+      throw MooseException("The provided boundary is not an open single-segment boundary.");
+      return false;
+    }
+    else
+      return true;
+  }
```

Comment:
other instances are good to go but this one still seems like it wont age well with new exceptions being added to the called routines. 

If you get an exception, and it s not the one you expect, then you return all good / true.